src/hil/src/spi.rs

```diff
@@ -108,7 +112,27 @@ pub trait SpiMaster {
     // is any of the read/read_write calls. These functions
     // allow an application to manually control when the
     // CS line is high or low, such that it can issue multi-byte
-    // requests with single byte oprations.
+    // requests with single byte operations.
     fn hold_low(&self);
     fn release_low(&self);
 }
+
+/// SPIMasterDevice provides a chip-specific interface to the SPI Master
+/// hardware. The interface wraps the chip select line so that chip drivers
+/// cannot communicate with different SPI devices.
+pub trait SPIMasterDevice {
```

Comment:
A slave will have only 1 CS line yes, but which GPIO it's assigned to can vary. It's the slave equivalent of https://github.com/helena-project/tock/pull/108/files#diff-a6e1a6aff89e187bbbd0a32d059c9f7bR97